src/ast.ls

```diff
@@ -2350,6 +2350,9 @@ class exports.For extends While
                 then "#idx #{ '<>'char-at pvar < 0 }#eq #tvar"
                 else "#pvar < 0 ? #idx >#eq #tvar : #idx <#eq #tvar"
         else
+            if @source instanceof Literal and @source.value is /^[0-9]+$/
```

Comment:
And yet `-42.length` triggers a syntax error anyway.

I'm starting to have misgivings about this approach; maybe I should be adding robustness (generating `(42).length`) instead of trying to catch the problem cases and raise errors? I was reluctant to do that at first because errors seem better than compiling things that I know are wrong, but this is turning into a slightly awkward patch. Do you have an opinion?